debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/VirtualThreadsCommand.java

```diff
@@ -0,0 +1,120 @@
+ * !continuationstack 0x00007fe78c244ac0 !j9vmcontinuation 0x00007fe78c244ac0 !j9object 0x00000007064250D8 (Continuation) !j9object 0x0000000706425038 (VThread) -
+ * .....
+ */
+public class VirtualThreadsCommand extends Command {
+	public VirtualThreadsCommand() {
+		addCommand("vthreads", "", "Lists virtual threads");
+	}
+
+	@Override
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException {
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			if (!JavaVersionHelper.ensureJava19AndUp(vm, out)) {
+				return;
+			}
+		} catch(CorruptDataException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+		displayVirtualThreads(out);
+	}
+
+	/**
+	 * Prints all the live virtual threads
+	 *
+	 * @param out the PrintStream to write output to
+	 */
+	private void displayVirtualThreads(PrintStream out) throws DDRInteractiveCommandException {
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			Class<J9JavaVMPointer> vmClass = J9JavaVMPointer.class;
+			Lookup lookup = MethodHandles.lookup();
+			MethodHandle liveVirtualThreadListMH = lookup.findVirtual(vmClass, "liveVirtualThreadList", MethodType.methodType(PointerPointer.class));
+			PointerPointer mainVirtualThread = (PointerPointer)liveVirtualThreadListMH.invoke(vm);
+
+			MethodHandle virtualThreadLinkNextOffsetMH = lookup.findVirtual(vmClass, "virtualThreadLinkNextOffset", MethodType.methodType(UDATA.class));
+			UDATA linkNextOffset = (UDATA)virtualThreadLinkNextOffsetMH.invoke(vm);
+
+			if (mainVirtualThread.notNull() && mainVirtualThread.at(0).notNull()) {
+				/*
+				 * liveVirtualThreadList is a doubly-linked-list storing all the live virtual threads
+				 * The root node is a dummy virtual thread marking the start and the end of the list
+				 */
+				J9ObjectPointer root = J9ObjectPointer.cast(mainVirtualThread.at(0));
+				J9ObjectPointer node = ObjectReferencePointer.cast(root.addOffset(linkNextOffset)).at(0);
+				J9ObjectFieldOffset nameOffset = J9ObjectHelper.getFieldOffset(node, "name", "Ljava/lang/String;");
+				J9ObjectFieldOffset contOffset = J9ObjectHelper.getFieldOffset(node, "cont", "Ljdk/internal/vm/Continuation;");
+				J9ObjectPointer cont = J9ObjectHelper.getObjectField(node, contOffset);
+				J9ObjectFieldOffset vmRefOffset = J9ObjectHelper.getFieldOffset(cont, "vmRef", "J");
+
+				while (!node.eq(root)) {
+					J9ObjectPointer name = J9ObjectHelper.getObjectField(node, nameOffset);
+					cont = J9ObjectHelper.getObjectField(node, contOffset);
+					long vmRef = J9ObjectHelper.getLongField(cont, vmRefOffset);
+					String formattedOutput = String.format(
+							"!continuationstack 0x%016x !j9vmcontinuation 0x%016x !j9object %s (Continuation) !j9object %s (VThread) - %s",
+									vmRef,
+									vmRef,
+									cont.getHexAddress(),
+									node.getHexAddress(),
+									J9ObjectHelper.stringValue(name));
```

Comment:
These lines should all be indented the same as the first (`"!continuationstack...`).